guardian/master-to-main
# refactor: Convert to ESM

## What does this change?

Converts the project to ESM from CommonJS. This allows us to keep our dependencies up to date, such as Ora which has become ESM only at v6 and which contains a high vulnerability in v5.

Specifically:
- The project config has been converted to ESM.
- The run script has been converted into ESM and in the process the dev script has been split out into its own file, following the [Oclif recommended pattern](https://oclif.io/docs/esm).
- Some tests that required mocking Ora were removed as after discussion this was the most efficient way forward, as it is hard to mock ESM modules in Jest. This repo has been in use for some time and is well tested, and the tests were only for the spinner.
- Ora has been bumped from v5 -> v6.
- Node has been bumped from v14 -> v16. 
- Jest dependencies have been bumped.

## How to test

On this branch, run `yarn run:dev  guardian/master-to-main-demo <YOUR PAT>` and see that it connects to the repo.

## How can we measure success?

We have fewer vulnerabilities in our dashboards.
